pkg/cluster/deploystorage.go

```diff
@@ -75,10 +76,23 @@ func (m *manager) ensureResourceGroup(ctx context.Context) error {
 	}
 
 	_, err := m.resourceGroups.CreateOrUpdate(ctx, resourceGroup, group)
-	if requestErr, ok := err.(*azure.RequestError); ok &&
-		requestErr.ServiceError != nil && requestErr.ServiceError.Code == "RequestDisallowedByPolicy" {
+
+	var serviceError *azure.ServiceError
+	// CreateOrUpdate wraps DetailedError wrapping a *RequestError (if error generated in ResourceGroup CreateOrUpdateResponder at least)
+	if detailedErr, ok := err.(autorest.DetailedError); ok {
+		if requestErr, ok := detailedErr.Original.(*azure.RequestError); ok {
+			serviceError = requestErr.ServiceError
+		}
+	}
+
+	// TODO [gv]: Keeping this for retro-compatibility, but probably this can be removed
+	if requestErr, ok := err.(*azure.RequestError); ok {
+		serviceError = requestErr.ServiceError
+	}
```

Comment:
+1 on this.  Note: this is identical to the following code in `func (m *manager) deployARMTemplate`:
```golang
	serviceErr, _ := err.(*azure.ServiceError) // futures return *azure.ServiceError directly

	// CreateOrUpdate() returns a wrapped *azure.ServiceError
	if detailedErr, ok := err.(autorest.DetailedError); ok {
		serviceErr, _ = detailedErr.Original.(*azure.ServiceError)
	}
```
Nothing to do about it now, but worth keeping an eye in case this becomes a wider pattern / best practice.